### PR TITLE
readme and docs updates

### PR DESCRIPTION
**Pull request by zackbradys**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
* README and Documentation updates.

**What is the current behavior?** (You can also link to an open issue here)
* N/A

**What is the new behavior (if this is a feature change)?**
* N/A

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
* No

**Other information**:
* N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
 
-As OCI registries have become ubiquitous nowadays for storing and distributing containers.  Their success and widespread adoption has led many projects to expand beyond containers.
+For more information, please review the **[Hauler Documentation](https://rancherfederal.github.io/hauler-docs)!**
 
-`hauler` capitalizes on this, and leverages the [`oci`](https://github.com/opencontainers) spec to be a simple, zero dependency tool to collect, transport, and distribute your artifacts.
+## Latest Release
+* [Hauler v0.3.0](https://github.com/rancherfederal/hauler/releases/tag/v0.3.0)
+  * [Download - Source](https://github.com/rancherfederal/hauler/archive/refs/tags/v0.3.0.zip)
+  * [Download - Linux AMD64](https://github.com/rancherfederal/hauler/releases/download/v0.3.0/hauler_0.3.0_linux_amd64.tar.gz)
+  * [Download - Linux ARM64](https://github.com/rancherfederal/hauler/releases/download/v0.3.0/hauler_0.3.0_linux_arm64.tar.gz)
```
Comment (dweomer) [inline README.md]: I don't think this kind of linkage is appropriate on a readme for anything with even semi-active development. It becomes yet another thing to have to update, and forget to do so, and thereby mislead customers/users.

File: README.md [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
 * [cosign](https://github.com/sigstore/cosign)
 
 ## Notices
+**WARNING - Upcoming Deprecated Command(s):**
 
-*** WARNING: Deprecated Command *** 
-
-The `hauler download (dl)` and `hauler serve (not the store version)` commands are deprecated and will be removed in a future release of Hauler.
\ No newline at end of file
+`hauler download (dl)` and `hauler serve (not the store version)` commands are deprecated and will be removed in a future release.
```
Comment (dweomer) [inline README.md]:
```suggestion
`hauler download` (alternatively, `dl`) and `hauler serve` (_not_ `hauler store serve`) commands are deprecated and will be removed in a future release.
```

File: README.md (hunk 1 above)
Comment (zackbradys) [reply README.md]: very fair point, users can easily access the releases page. will fix and remove this section in the readme!

**Review by zackbradys — COMMENTED** [review]
LGTM!

**Pull request merged**